op-conductor/conductor/service.go

```diff
@@ -815,9 +902,31 @@ func (oc *OpConductor) startSequencer() error {
 			oc.log.Warn("sequencer already started.", "err", err)
 		}
 	}
-	oc.metrics.RecordStartSequencer(err == nil)
 
+	sequencerStarted = true
+
+	// Set up deferred cleanup in case of errors with rollup boost
+	defer func() {
+		if sequencerStarted && !oc.seqActive.Load() && oc.ctrl != nil {
+			// If we started the sequencer but failed to set rollup boost (and thus didn't set seqActive),
+			// try to stop the sequencer again
+			_, stopErr := oc.ctrl.StopSequencer(oc.shutdownCtx)
+			if stopErr != nil {
+				oc.log.Error("failed to roll back sequencer state after rollup boost failure", "err", stopErr)
+			} else {
+				oc.log.Warn("sequencer rolled back due to rollup boost failure")
+			}
+		}
+	}()
+
+	// Enable rollup boost
+	if err := oc.rollupBoost.SetExecutionMode(ctx, client.ExecutionModeEnabled); err != nil {
```

Comment:
rollup boost has to be started before sequencer started, otherwise sequencer may not be able to send Forkchoiceupdate call to builder

Reply:
1. we can't start sequencer if we cannot start rollupboost.
2. if start sequencer failed, we need to stop rollupboost